src/main/java/org/jenkinsci/plugins/ssegateway/sse/EventDispatcher.java

```diff
@@ -446,6 +454,9 @@ synchronized void processRetries() {
             // in my opinon the statement inside the finally should be
             // sufficient - but without this second false i had some
             // endless loops
+            if (!retryQueue.isEmpty()) {
+                scheduleRetryQueueProcessing(RETRY_QUEUE_PROCESSING_DELAY);
```

Comment:
The bottom `isRetryLoopActive` was replicated replicating the retry scheduling to ensure in case the scenario in the comment happens retries still get scheduled

Reply:
You and me are aligned on this, This entire section isn't needed, maybe it should just be removed but apparently the original author had issues